Add cloud-run documentation for testing framework (#33875)
hashicorp/terraform: website/docs/cli/commands/test.mdx

The following options apply to the Terraform `test` command:

> * `-cloud-run=<module source>` - This test run will execute remotely on Terraform Cloud within the specified Terraform [private registry](/terraform/language/modules/sources#terraform-registry) module.

Reviewer: ```suggestion
* `-cloud-run=<module source>` - This test run executes remotely on Terraform Cloud within the specified Terraform [private registry](/terraform/language/modules/sources#terraform-registry) module.
```

You should monitor the output of the test command closely to ensure Terraform removes the infrastructure it created or perform manual cleanup if not. We recommend creating dedicated testing accounts within the target providers that you can routinely and safely purge to ensure any accidental and costly resources aren't left behind.

Terraform also provides diagnostics explaining why it could now automatically clean up. You should resolve these diagnostics to ensure that future clean-up operations are successful.

> ## Terraform Cloud Execution

Reviewer: sentence case:
```suggestion
## Terraform Cloud execution
```

> You can execute your tests remotely on Terraform Cloud using the `-cloud-run` option.

Reviewer: ```suggestion
You can execute tests remotely on Terraform Cloud using the `-cloud-run` option.
```

> The argument accepts a private registry module source as specified in the [Terraform Registry Module Sources](/terraform/language/modules/sources#terraform-registry) documentation. This will associate the test run with the given private module within the Terraform Cloud UI.

Reviewer: Can we shorten this to:
```suggestion
The `-cloud-run` command accepts a [private registry module source](/terraform/language/modules/sources#terraform-registry) as as argument. This argument associates the test run with your specified private module within the Terraform Cloud user interface.
```

Author: Tweaked this a bit more to get around confusino of command vs argument vs option:

```
The `-cloud-run` option accepts a [private registry module source](/terraform/language/modules/sources#terraform-registry). This option associates the test run with your specified private module within the Terraform Cloud user interface.
```

> You must provide a module from a **private** registry and not the public Terraform registry.

Reviewer: I like the bold, but we use italics for emphasis, so I think we should change to:
```suggestion
You must provide a module from a _private_ registry, not the public Terraform registry.
```

> You must execute [`terraform login`](/terraform/cli/command/login) before using this option, and ensure that the provided `hostname` argument matches the private registry hostname of your target module.

Reviewer: ```suggestion
You must execute [`terraform login`](/terraform/cli/command/login) before using this option, and ensure that your `hostname` argument matches the private registry hostname of your target module.
```
